### PR TITLE
Updates to Drawer styles

### LINKED ISSUE #145: Drawer should auto expand on activeItem
#### Describe the bug
drawer does not expand automatically when navigating inside docit

#### What is the expected behavior?
drawer should expand, not just on the first run

#### What are the steps to reproduce?
1. Go to home page
2. Click on any internal link
3. See the drawer on the left that doesn't expand

#### What is your environment?
 Browser
- [ ] Internet Explorer
- [ ] Edge
- [ ] Safari
- [x] Chrome
- [ ] Firefox

 Operating System
- [x] MacOS
- [ ] Windows
- [ ] Other

### LINKED ISSUE #215: DrawerLayout shall have access to drawer's expanded/collapsed state
#### Describe the desired behavior

DrawerLayout shall know when a drawer is expanded or when a drawer is collapsed

#### Describe the current behavior

It can only guess the drawer state by comparing the shadows

#### Is this request related to a current issue?

#201 , #210 , and PR #214

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-1515

Fixes #215, fixes #145

#### Changes proposed in this Pull Request:
- Add 4dp shadow to drawer by default 
    - Can be turned off and replaced with a border by using `sideBorder={true}`
- Use fontWeight 400 for all inactive nav item titles
- Use font-weight 600 for active item title
- Turn off NavItem dividers by default
- Use square highlight for active items by default
    - Use primary[500] with 5% opacity for highlight color
- Use overline variant by default for NavGroup titles
- Add `hideContentOnCollapse` prop to Header and Footer (enabled by default)
- Use font-weight 600 for any items in the selected item's tree hierarchy (can be turned off with `disableActiveItemParentStyles={true}` on the drawer, drawerBody, or DrawerNavGroup.

> TODO: Need to update the storybook and some of the docs

> **NOTE**: Some of these changes accidentally got pushed directly to dev, so in testing make sure to check all points above since some of them will not be reflected in the diff. (You can look at a diff with master if you want to see the code changes).

#### Screenshots:
image

### CODE REVIEW

**Comment by huayunh** [comment]
Drawer's shadow should cast on the app bar, not app bar shadow casting on the drawer (is this a drawerLayout thing?)

**Comment by joebochill** [comment]
> Drawer's shadow should cast on the app bar, not app bar shadow casting on the drawer (is this a drawerLayout thing?)

It's a z-index thing...I'll update the DrawerLayout

**Comment by huayunh** [comment]
Would like to see the divider turned off by default on line 169 of `/components/src/core/Drawer/DrawerNavGroup.tsx`

File: components/src/core/Drawer/DrawerNavItem.tsx [hunk 1]
```diff
@@ -143,8 +145,24 @@ const DrawerNavItemRender: React.ForwardRefRenderFunction<unknown, DrawerNavItem
         .fade(0.8)
         .rgb()
         .string();
+    // approximating primary[200] but we don't have access to it directly from the theme
+    const lightenedPrimary = color(theme.palette.primary.main)
+        .lighten(0.83)
+        .desaturate(0.39)
+        .rgb()
```
Comment (huayunh) [inline components/src/core/Drawer/DrawerNavItem.tsx:152]: `.rgb()` before `.string()` is not necessary

**Comment by joebochill** [comment]
> Would like to see the divider turned off by default on line 169 of `/components/src/core/Drawer/DrawerNavGroup.tsx`

I think we had discussed this before and decided to leave it in at the time. If we turn it off, there's currently no way to turn it back on unless we use the existing divider prop...but then what if you want a divider on the group titles but not on the nav items — we'd need another prop. I'm open to the idea, but I would make a new issue to discuss it rather than doing it in this PR since it wasn't part of the initial scope.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+-   Updated default styles for `<Drawer>`
 
-### Fixed
-
--   `<Drawer>` nested nav item dark-themed background color.
+### Added
+-   Additional configuration properties for `<Drawer>`
+    -   `sideBorder` gives the `<Drawer>` a slight border instead of a drop shadow
+    -   `disableActiveItemParentStyle` disables the bold text style for active item's parent elements
+-   Additional configuration options for `<DrawerHeader>` and `<DrawerFooter>`
```
Comment (emclaug2) [inline CHANGELOG.md:14]: `hideContentOnCollapse` was added to `<DrawerSubHeader>` not `<DrawerHeader>`, right?

**Review by emclaug2 — APPROVED** [review]
Changes work nicely! Just found one thing in the Drawer docs.

**Comment by huayunh** [comment]
took me around 20 min to figure out what that `notifyActiveParent` is doing — pretty clever :) 

everything looks good design-wise. Just waiting for that docit thing to get fixed?

**Comment by joebochill** [comment]
> took me around 20 min to figure out what that `notifyActiveParent` is doing — pretty clever :)
 
Took me more than 20 minutes to figure out how to make it work

**Comment by huayunh** [comment]
Works great on docit after the updates

**Pull request merged**